### PR TITLE
Rework content disposal system

### LINKED ISSUE #127: 🚧Rework content disposal system
### I have done the items below . . .

- [X] I have updated the title by replacing the '**_<title_**>' section.

### Description

When disposing of `IContent` items during the game loop, the app throws an **_OpenGL_** exception.  This is due to the fact that during the disposal process, the GPU side of things are destroyed which is what is supposed to happen.  The problem is that when the batch is ended and Velaptor tells _**OpenGL**_ to render what is in the batch, the texture has been destroyed which is why an _**OpenGL**_ error occurs.

The idea is to remove the `Dispose()` method and pattern from `IContent` items, which forces the user to have to use the `IContentLoader` implementation to unload content.

To prevent an _**OpenGL**_ error from occurring, the user calls `IContentLoader.Unload()`, which will send an `Observable` notification that will remove the batch item immediately upon disposal to prevent an attempt to render it when the batch ends.

This means that the batcher (`TextureBatchService`), as well as other batchers, will have to subscribe to the **_Observable_** to receive a notification from `IContentLoader` with a special ID to find the batch item so it can be removed.

Create an `Observable` that will replace the `Dispose()` methods in each `IContentLoader` implementation.  These methods are only meant to be called once by _**Velaptor**_ to be able to free resources when the application is shut down.  Currently the `Dispose()` method can be called anytime by the user of the library which means this will unload the content system as a whole which is only meant to occur when the app is closed.  This will crash the application and needs to be prevented.

### Acceptance Criteria

**This issue is finished when:**
- [x] Remove `IsPooled` property from `IContent`
- [x] Create a new `Observable` called `RemoveBatchItemObservable`
    - Purpose: Used to remove a batch item in the `TextureBatchService`
    - [x] Make this class a singleton which will be instantiated from the **IoC Container** upon creation
    - [x] Will send out the texture ID in the broadcast so that way the proper batch item can be found for removal
- [x] Create a new `Observable` called `ShutDownObservable`
    - Purpose: Used to signal that the application is shutting down and all content needs to be disposed of.  This will send out `Dispose` observable notifications to dispose of all the content items.
    - [x] Make this class a singleton which will be instantiated from the **IoC Container** upon creation
    - [x] Setup each `ILoader` implementation to subscribe to this observable to get a notification to shut down by unloading all of the currently loaded content and remove disposal code
        - [x] AtlasLoader
        - [x] FontLoader
        - [x] SoundLoader
        - [x] TextureLoader
- [x] Create a new `Observable` called `DisposeObservable`
    - Purpose: Disposes of a content item.  This replaces the public `IDisposable` implementations to prevent user from invoking them.
    - [x] Make this class a singleton which will be instantiated from the **IoC Container** upon creation
    - [x] Setup each `ITexture` implementation to subscribe to this observable to get a notification to dispose of resources
    - [x] Unsubscribe from the observable when the observable notification is received
- [x] Setup `SpriteBatch` implementation
    - [x] Remove `IDisposable` interface from `ISpriteBatch` interface
    - [x] Inject `ShutDownObservable`
    - [x] Move disposal code to subscription method
- [x] Setup `IShaderProgram` implementations
    - [x] Remove `IDispose` from `IShaderProgram` interface
    - [x] `TextureShader`
        - [x] Remove `IDisposable` interface from `IShaderProgram` interface
        - [x] Inject `ShutDownObservable`
        - [x] Move disposal code to subscription method
    - [x] `FontShader`
        - [x] Inject `ShutDownObservable`
        - [x] Move disposal code to subscription method
- [x] Setup `IGPUBuffer<T>` implementations
    - [x] Remove `IDispose` from `IGPUBuffer<T>` interface
    - [x] `TextureBuffer`
        - [x] Remove `IDisposable` interface from `IGPUBuffer<T>` interface
        - [x] Inject `ShutDownObservable`
        - [x] Move disposal code to subscription method
    - [x] `FontBuffer`
        - [x] Inject `ShutDownObservable`
        - [x] Move disposal code to subscription method

_**NOTE: It was decided to no implement these due to future changes**_
- ~☑️ Setup the each UI control implementation to send out a notification to the `TextureBatchService` using the `RemoveBatchItemObservable` to remove any pending batch items that will be rendered~
    - ~☑️ `Label`~
    - ~☑️ `Button`~
- ~☑️ Setup `TextureBatchService`~
    - ~☑️ Inject `RemoveBatchItemObservable` observable for subscription and use~
    - ~☑️ The notification will come with the **_Texture ID_** identify which batch item to remove~
        - ~🗒️The idea of "removing" a batch item is not actually physically removing the item from the batch list, but simply settings its `shouldRender` to a value of false and calling its `Empty()` method.~
    - ~☑️ Disposing of the service will unsubscribe from the observable~
- [x] Code documentation added if required

### ToDo Items

- [x] Draft pull request created and linked to this issue
- [X] Priority label added to issue (**_low priority_**, **_medium priority_**, or **_high priority_**)
- [X] Issue linked to the proper project
- [X] Issue linked to proper milestone
- [x] Unit tests have been written and/or adjusted for code additions or changes
- [x] All unit tests pass

### Issue Dependencies

_No response_

### Related Work

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### LINKED ISSUE #148: 🧪QA Testing Velaptor
### To Do List:

- [X] I have linked all issues that are getting tested for this release
- [X] QA testing label added to issue (🧪qa testing)

### Version Of Velaptor Being Tested

v1.0.0-preview.6

### Relevant Information

_No response_

### Issues To Test

- [x] Just make sure that the application runs normally.

### QA Testing Dependencies

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by CalvinWilkinson**
<h1 style="font-weight:bold" align="center">Preview Feature Pull Request</h1>

<details><summary>📄Description📄</summary>

This pull request is for preview release **_v1.0.0-preview.6_**
</details>

<h2 style="font-weight:bold" align="center">✅Development Checklist✅</h2>

<details open><summary>🌳Branching🌳</summary>
The name of the preview feature branch for this pull request has the following syntax.

Syntax: _preview/feature/&lt;issue-id&gt;-&lt;description&gt;_  
Example: _preview/feature/123-my-preview-change_
- [x] Yes
- [ ] No

The name of the branch for this pull request is merging into a preview branch with the following syntax.

Syntax: _preview/v&lt;major-number&gt;.&lt;minor-number&gt;.&lt;patch-number&gt;-preview.&lt;prev-number&gt;_  
Example: _preview/v1.2.3-preview.4_
- [x] Yes
- [ ] No

The preview feature branch for this pull request was created from a preview branch with the following syntax.

Syntax: _preview/feature/&lt;issue-id&gt;-&lt;description&gt;_  
Example: _preview/feature/123-my-preview-change_
- [x] Yes
- [ ] No
</details>

<details open><summary>🐛Bugs🐛</summary>

Contains Bug Fix(es)
- [ ] Yes
    - [ ] A bug-label label has been added to the pull request.
- [x] No
</details>

<details open><summary>💣Breaking Change(s)💣</summary>

Any changes, including behavioral, that prevent a library user's application from compiling or behaving correctly.
Refer to this [link](https://docs.microsoft.com/en-us/dotnet/core/compatibility/#modifications-to-the-public-contract) for more information.
- [x] Yes
- [ ] No
</details>

<details open><summary>✨Enhancements✨</summary>

Contains enhancements that add a feature or behavior.
- [x] Yes
    - [x] An enhancement-label label has been added to the pull request.
- [ ]  No
</details>

<details open><summary>⚙️Workflow (CI/CD) Changes⚙️</summary>

These kinds of changes are only done by the project owner and maintainers.
- [ ] Yes
    - [ ] A workflow-label label has been added to the pull request.
- [x] No
</details>

<details open><summary>📃Documentation Updates📃</summary>

Contains changes that require documentation updates to code docs or **Velaptor** documentation.
- [x] Yes
    - [ ] I have updated the documentation accordingly.
    - [x] A documentation-label label has been added to the pull request.
- [ ] No
</details>

<details open><summary>🧪Unit Testing🧪</summary>

My change requires unit tests to be written.
- [x] Yes
    - [x] I have added tests to cover my changes.
- [ ] No
</details>

<details open><summary>🧪Manual Testing🧪</summary>

I have manually tested my changes to the best of my ability.
This can be done by using the included testing application.
- [x] Yes
- [ ] No
</details>

<h2 style="font-weight:bold" align="center">✅Code Review Checklist✅</h2>

- [x] Pull request title matches the title of the linked issue.
- [x] The **_[add version here]_** text in the pull request description replaced with the version.
- [x] Associated issue exists and is linked to this pull request.
    - One issue per pull request.
- [x] My code follows the coding style of this project.
    - This is enforced by the *.editorconfig* files in the project and displayed as warnings.  If there is an edge case with coding style that should be ignored or changed, reach out and lets discuss it.
- [x] All tests passed locally.
    - Status checks are put in place to run unit tests every single time a change is pushed to a pull request.  This does not mean that the tests pass in both the local and CI environment.
- [x] A preview-label label has been added to the pull request.

### CODE REVIEW

File: Testing/VelaptorTests/OpenGL/GLWindowTests.cs [hunk 1]
```diff
@@ -170,9 +169,29 @@ public void Ctor_WithNullContentLoader_ThrowsException()
                     this.mockPlatform.Object,
                     this.mockTaskService.Object,
                     null,
-                    this.mockGLObservable.Object);
+                    this.mockGLInitReactor.Object);
             }, "The parameter must not be null. (Parameter 'contentLoader')");
         }
+
+        [Fact]
+        public void Ctor_WithNullGlInitReactor_ThrowsException()
```
Comment (CalvinWilkinson) [inline Testing/VelaptorTests/OpenGL/GLWindowTests.cs:177]: Refactor to below.  

```suggestion
        public void Ctor_WithNullGLInitReactor_ThrowsException()
```

File: Velaptor/Content/Caching/IDisposableItemCache.cs [hunk 2]
```diff
@@ -6,6 +6,8 @@ namespace Velaptor.Content.Caching
 {
     using System;
 
+    // TODO: Remove this
```
Comment (CalvinWilkinson) [inline Velaptor/Content/Caching/IDisposableItemCache.cs:9]: Delete entire file

File: Velaptor/Content/Caching/TextureCache.cs [hunk 3]
```diff
@@ -51,25 +55,48 @@ internal sealed class TextureCache : IDisposableItemCache<string, ITexture>
         /// <param name="fontAtlasService">Provides font atlas services.</param>
         /// <param name="fontMetaDataParser">Parses metadata that might be attached to the file path.</param>
         /// <param name="path">Provides path related services.</param>
+        /// <param name="shutDownReactor">Sends push notifications that the application is shutting down.</param>
+        /// <param name="disposeTexturesReactor">Sends a notifications to dispose of textures.</param>
```
Comment (CalvinWilkinson) [inline Velaptor/Content/Caching/TextureCache.cs:59]: Refactor to below

```suggestion
        /// <param name="disposeTexturesReactor">Sends push notifications to dispose of textures.</param>
```

File: Velaptor/Content/Factories/SoundFactory.cs [hunk 4]
```diff
@@ -16,7 +21,75 @@ namespace Velaptor.Content.Factories
...
     {
+        private static readonly Dictionary<uint, string> Sounds = new ();
+        private readonly IDisposable disposeSoundUnsubscriber;
+        private readonly IDisposable shutDownUnsubscriber;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SoundFactory"/> class.
+        /// </summary>
+        /// <param name="disposeSoundReactor">Sends push notifications to dispose of sounds.</param>
+        /// <param name="shutDownReactor">Sends push notifications that the application is shutting down.</param>
```
Comment (CalvinWilkinson) [inline Velaptor/Content/Factories/SoundFactory.cs:32]: Refactor to below

```suggestion
        /// <param name="shutDownReactor">Sends a push notification that the application is shutting down.</param>
```

File: Velaptor/Content/Fonts/IFont.cs [hunk 5]
```diff
@@ -38,6 +40,12 @@ public interface IFont : IContent
         /// </summary>
         VelFontStyle Style { get; set; }
 
+        /// <summary>
+        /// Gets a value indicating whether or not font is a default font.
```
Comment (CalvinWilkinson) [inline Velaptor/Content/Fonts/IFont.cs:44]: Refactor below

```suggestion
        /// Gets a value indicating whether or not the font is a default font.
```

File: Velaptor/Content/Texture.cs [hunk 6]
```diff
@@ -74,24 +118,21 @@ internal Texture(IGLInvoker gl, IGLInvokerExtensions glExtensions, string name,
         /// <inheritdoc/>
         public uint Height { get; private set; }
 
-        /// <inheritdoc/>
-        public bool IsDisposed { get; private set; }
-
-        /// <inheritdoc cref="IDisposable.Dispose"/>
-        public void Dispose()
+        /// <summary>
+        /// Disposes of the texture if this textures <see cref="Id"/> matches the given <paramref name="data"/>.
```
Comment (CalvinWilkinson) [inline Velaptor/Content/Texture.cs:122]: Refactor to below

```suggestion
        /// Disposes of the texture if this textures <see cref="Id"/> matches the texture ID in the given <paramref name="data"/>.
```

File: Velaptor/Content/Texture.cs [hunk 7]
```diff
@@ -74,24 +118,21 @@ internal Texture(IGLInvoker gl, IGLInvokerExtensions glExtensions, string name,
...
+        /// </summary>
+        /// <param name="data">The ID of the texture to dispose.</param>
```
Comment (CalvinWilkinson) [inline Velaptor/Content/Texture.cs:124]: Refactor to below

```suggestion
        /// <param name="data">The data of the texture to dispose.</param>
```

File: Velaptor/IoC.cs [hunk 8]
```diff
@@ -3,6 +3,8 @@
 // </copyright>
 
 using System.Runtime.CompilerServices;
+using Velaptor.Observables.ObservableData;
```
Comment (CalvinWilkinson) [inline Velaptor/IoC.cs:6]: Move this using statement to the rest of the using statements below

File: Velaptor/Observables/DisposeSoundsReactor.cs [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+{
+    // ReSharper disable RedundantNameQualifier
+    using System.Diagnostics.CodeAnalysis;
+    using Velaptor.Observables.Core;
+    using Velaptor.Observables.ObservableData;
+
+    // ReSharper restore RedundantNameQualifier
+
+    /// <summary>
+    /// Creates an observable to send push notifications to signal that an object needs to be disposed.
```
Comment (CalvinWilkinson) [inline Velaptor/Observables/DisposeSoundsReactor.cs:15]: Refactor to below

```suggestion
    /// Creates an observable to send push notifications to signal that a sound needs to be disposed.
```

File: Velaptor/Observables/DisposeSoundsReactor.cs [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+    /// </summary>
+    internal class DisposeSoundsReactor : Reactor<DisposeSoundData>
+    {
+        /// <summary>
+        /// Sends a push notification to signal that an object needs to be disposed.
```
Comment (CalvinWilkinson) [inline Velaptor/Observables/DisposeSoundsReactor.cs:20]: Refactor to below

```suggestion
        /// Sends a push notification to dispose of a sound.
```

File: Velaptor/Observables/DisposeTexturesReactor.cs [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+{
+    // ReSharper disable RedundantNameQualifier
+    using System.Diagnostics.CodeAnalysis;
+    using Velaptor.Observables.Core;
+    using Velaptor.Observables.ObservableData;
+
+    // ReSharper restore RedundantNameQualifier
+
+    /// <summary>
+    /// Creates an observable to send push notifications to signal that an object needs to be disposed.
```
Comment (CalvinWilkinson) [inline Velaptor/Observables/DisposeTexturesReactor.cs:15]: Refactor to below

```suggestion
    /// Creates an observable to send push notifications to signal that a texture needs to be disposed.
```

File: Velaptor/Observables/DisposeTexturesReactor.cs [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+    /// </summary>
+    internal class DisposeTexturesReactor : Reactor<DisposeTextureData>
+    {
+        /// <summary>
+        /// Sends a push notification to signal that an object needs to be disposed.
```
Comment (CalvinWilkinson) [inline Velaptor/Observables/DisposeTexturesReactor.cs:20]: Refactor to below

```suggestion
        /// Sends a push notification to dispose of a texture.
```

File: Velaptor/Observables/ObservableData/GLInitData.cs [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+namespace Velaptor.Observables.ObservableData
+{
+    // ReSharper disable RedundantNameQualifier
+    using System.Diagnostics.CodeAnalysis;
+    using Velaptor.Observables.Core;
+
+    // ReSharper restore RedundantNameQualifier
+
+    /// <summary>
+    /// Holds da ta for the <see cref="IReactor{T}"/>.
```
Comment (CalvinWilkinson) [inline Velaptor/Observables/ObservableData/GLInitData.cs:14]: Fix spelling mistake

```suggestion
    /// Holds data for the <see cref="IReactor{T}"/>.
```

File: Velaptor/Observables/ObservableData/RemoveBatchItemData.cs [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+namespace Velaptor.Observables.ObservableData
+{
+    // ReSharper disable RedundantNameQualifier
+    using System.Diagnostics.CodeAnalysis;
+    using Velaptor.Observables.Core;
+
+    // ReSharper restore RedundantNameQualifier
+
+    /// <summary>
+    /// Holds da ta for the <see cref="IReactor{T}"/>.
```
Comment (CalvinWilkinson) [inline Velaptor/Observables/ObservableData/RemoveBatchItemData.cs:14]: Fix spelling mistake

```suggestion
    /// Holds data for the <see cref="IReactor{T}"/>.
```

File: Velaptor/Observables/ObservableData/ShutDownData.cs [hunk 15]
```diff
@@ -0,0 +1,20 @@
...
+namespace Velaptor.Observables.ObservableData
+{
+    // ReSharper disable RedundantNameQualifier
+    using System.Diagnostics.CodeAnalysis;
+    using Velaptor.Observables.Core;
+
+    // ReSharper restore RedundantNameQualifier
+
+    /// <summary>
+    /// Holds da ta for the <see cref="IReactor{T}"/>.
```
Comment (CalvinWilkinson) [inline Velaptor/Observables/ObservableData/ShutDownData.cs:14]: Fix spelling mistake

```suggestion
    /// Holds data for the <see cref="IReactor{T}"/>.
```

File: Velaptor/Observables/OpenGLContextReactor.cs [hunk 16]
```diff
@@ -1,28 +1,33 @@
...
 namespace Velaptor.Observables
 {
     // ReSharper disable RedundantNameQualifier
     using Velaptor.Observables.Core;
+    using Velaptor.Observables.ObservableData;
 
     // ReSharper restore RedundantNameQualifier
 
     /// <summary>
     /// Creates an observable to send push notifications of OpenGL events.
```
Comment (CalvinWilkinson) [inline Velaptor/Observables/OpenGLContextReactor.cs:14]: Refactor to below.

```suggestion
    ///  Creates an observable to send push notification that the <c>OpenGL</c> context has been created.
```

File: Velaptor/Observables/OpenGLContextReactor.cs [hunk 17]
```diff
@@ -1,28 +1,33 @@
...
     /// </summary>
-    internal class OpenGLContextObservable : Observable<object>
+    internal class OpenGLContextReactor : Reactor<GLContextData>
     {
         /// <summary>
         /// Sends a push notification that the OpenGL context has been created.
         /// </summary>
         /// <param name="data">The data to send with the notification.</param>
```
Comment (CalvinWilkinson) [inline Velaptor/Observables/OpenGLContextReactor.cs:21]: Add a summary section that the reactor(observable) auto unsubscribes itself after processing all of the push notifications.

File: Velaptor/Observables/OpenGLContextReactor.cs [hunk 18]
```diff
@@ -1,28 +1,33 @@
...
-        public virtual void OnGLContextCreated(object data)
+        public override void PushNotification(GLContextData data)
         {
-            foreach (var observer in Observers)
+            /* Work from the end to the beginning of the list
+               just in case the observable is disposed(removed)
+               in the OnNext() method.
+             */
+            for (var i = Observers.Count - 1; i >= 0; i--)
```
Comment (CalvinWilkinson) [inline Velaptor/Observables/OpenGLContextReactor.cs:28]: Add code after the for loop to unsubscribe all of the observers.  This will make the Reactor auto unsubscribing

File: Velaptor/Observables/OpenGLInitReactor.cs [hunk 19]
```diff
@@ -0,0 +1,39 @@
...
+{
+    // ReSharper disable RedundantNameQualifier
+    using System.Diagnostics.CodeAnalysis;
+    using Velaptor.Observables.Core;
+    using Velaptor.Observables.ObservableData;
+
+    // ReSharper restore RedundantNameQualifier
+
+    /// <summary>
+    /// Creates an observable to send push notifications of OpenGL events.
```
Comment (CalvinWilkinson) [inline Velaptor/Observables/OpenGLInitReactor.cs:15]: Refactor summary code doc

```suggestion
    /// Creates an observable to send push notification that OpenGL is initialized.
```

File: Velaptor/Observables/OpenGLInitReactor.cs [hunk 20]
```diff
@@ -0,0 +1,39 @@
...
+    /// </summary>
+    internal class OpenGLInitReactor : Reactor<GLInitData>
+    {
+        /// <summary>
+        /// Sends a push notification that OpenGL has been initialized.
+        /// </summary>
+        /// <param name="data">The data to send with the notification.</param>
```
Comment (CalvinWilkinson) [inline Velaptor/Observables/OpenGLInitReactor.cs:22]: Add summary section explaining that this Reactor unsubscribes all observers after processing push notifications if the new default param of `bool unsubscribeAfterProcessing` is true.

File: Velaptor/Observables/OpenGLInitReactor.cs [hunk 21]
```diff
@@ -0,0 +1,39 @@
...
+        [SuppressMessage(
+            "ReSharper",
+            "ForCanBeConvertedToForeach",
+            Justification = "Required for proper observable operation.")]
+        public override void PushNotification(GLInitData data)
```
Comment (CalvinWilkinson) [inline Velaptor/Observables/OpenGLInitReactor.cs:27]: Add new parameter called `bool unsubscribeAfterProcessing` with a default value of false

Do this for all `Reactor` types in code base

File: Velaptor/Observables/RemoveBatchItemReactor.cs [hunk 22]
```diff
@@ -0,0 +1,39 @@
...
+
+    // ReSharper restore RedundantNameQualifier
+
+    /// <summary>
+    /// Creates an observable to send push notifications to remove items from a batching service.
+    /// </summary>
+    internal class RemoveBatchItemReactor : Reactor<RemoveBatchItemData>
+    {
+        /// <summary>
+        /// Sends a push notification to remove a batch item from a batch service.
```
Comment (CalvinWilkinson) [inline Velaptor/Observables/RemoveBatchItemReactor.cs:20]: Refactor code doc

```suggestion
        /// Sends a push notification to remove a batch item.
```

File: Velaptor/OpenGL/GLWindow.cs [hunk 23]
```diff
@@ -52,7 +51,7 @@ internal sealed class GLWindow : IWindow
         /// <param name="platform">Information about the platform that is running the application.</param>
         /// <param name="taskService">Runs asynchronous tasks.</param>
         /// <param name="contentLoader">Loads various kinds of content.</param>
-        /// <param name="glObservable">Provides push notifications to OpenGL related events.</param>
+        /// <param name="glInitReactor">Provides push notifications to OpenGL related events.</param>
```
Comment (CalvinWilkinson) [inline Velaptor/OpenGL/GLWindow.cs:54]: Refactor doc

```suggestion
        /// <param name="glInitReactor">Provides push notifications that OpenGL has been initialized.</param>
```

**Review by kselena — CHANGES_REQUESTED** [review]

File: Testing/VelaptorTests/Content/Caching/TextureCacheTests.cs [hunk 24]
```diff
@@ -400,16 +477,45 @@ public void Unload_WhenItemToUnloadExists_RemovesAndDisposesOfTexture()
...
+            cache.Unload("non-existing-texture");
+
+            // Assert
+            this.mockDisposeTextureReactor
+                .Verify(m
+                    => m.PushNotification(new DisposeTextureData(123u)), Times.Never);
+        }
+
+        [Fact]
+        public void ShutDownNotification_WhenReceived_DisposesOfTextures()
```
[the remark this answers is not in the document]
Comment (CalvinWilkinson) [reply Testing/VelaptorTests/Content/Caching/TextureCacheTests.cs:508]: This is a special case where I am checking for behavior by triggering an event internal to the class.  I can't test it by executing a public method or property.  This is why you dont see a method name called `ShutDownNotification()`

**Review by kselena — APPROVED** [review]
